client/packages/system/src/Patient/VaccinationCards/VaccineCardTable.tsx

```diff
@@ -0,0 +1,142 @@
+import { VaccinationCardItemFragment } from '../api/operations.generated';
+
+interface VaccinationCardProps {
+  programEnrolmentId: string;
+  openModal: (
+    vaccinationId: string | null | undefined,
+    vaccineCourseDoseId: string
+  ) => void;
+  encounterId?: string;
+}
+
+const useStyleRowsByStatus = (rows?: VaccinationCardItemFragment[]) => {
+  const { setRowStyles } = useRowStyle();
+  const theme = useTheme();
+
+  useEffect(() => {
+    if (!rows) return;
+
+    const doneRows = rows.filter(row => row.given).map(row => row.id);
+    const notDoneRows = rows.filter(row => !row.given).map(row => row.id);
+
+    setRowStyles(doneRows, {
+      backgroundColor: `${theme.palette.background.success} !important`,
+    });
+    setRowStyles(
+      notDoneRows,
+      {
+        backgroundColor: 'white !important',
+      },
+      // Parameter to prevent the previous setRowStyles from being
+      // reset/overwritten
+      false
+    );
+  }, [rows]);
+};
+
+export const VaccinationCardComponent: FC<VaccinationCardProps> = ({
+  programEnrolmentId,
+  openModal,
+  encounterId,
+}) => {
+  const t = useTranslation('dispensary');
+  const { localisedDate } = useFormatDateTime();
+  const theme = useTheme();
+
+  const {
+    query: { data, isLoading },
+  } = usePatientVaccineCard(programEnrolmentId);
+
+  useStyleRowsByStatus(data?.items);
+
+  const columns = useColumns<VaccinationCardItemFragment>([
+    {
+      key: 'age',
+      label: 'label.age',
+      sortable: false,
+      accessor: ({ rowData }) =>
+        t('label.age-months-count', { count: rowData.minAgeMonths }),
+    },
+    {
+      key: 'label',
+      label: 'label.dose',
+      accessor: ({ rowData }) => rowData.label,
+    },
+    {
+      key: 'status',
+      label: 'label.status',
+      accessor: ({ rowData }) => rowData.given,
+      Cell: ({ ...props }) => (
+        <StatusCell
+          {...props}
+          statusMap={{
+            true: {
+              color: theme.palette.success.light,
+              label: t('label.status-given'),
+            },
+            false: {
+              color: theme.palette.error.main,
+              label: t('label.status-not-given'),
+            },
+          }}
+        />
+      ),
+    },
+    {
+      key: 'suggestedDate',
+      label: 'label.suggested-date',
+      accessor: ({ rowData }) => localisedDate(rowData.suggestedDate ?? ''),
+    },
+    {
+      key: 'dateGiven',
+      label: 'label.date-given',
+      accessor: ({ rowData }) => localisedDate(rowData.vaccinationDate ?? ''),
+    },
+  ]);
+
+  return isLoading ? (
+    <InlineSpinner />
+  ) : (
+    <>
+      <DataTable
+        id={'Vaccine Course List'}
+        columns={columns}
+        data={data?.items ?? []}
+        isLoading={isLoading}
+        onRowClick={row =>
+          (encounterId ?? row.vaccinationId)
+            ? openModal(row.vaccinationId, row.vaccineCourseDoseId)
+            : undefined
+        }
```

Comment:
One thing I don't like about this -- even though you can't open the Patient Vax card non-vaccination rows, you still get the "clickable" icon when hovering over the rows. The only way to prevent this is to not have return a function here for those rows, but the problem is we need access to the *row* data in order to determine the state, which we can only do via a function. 🤷‍♂️

Reply:
Oof yeah, that's frustrating... maybe we can do something in the setRowStyles, for rows that don't have a vaccinationId?

```css
        '& td': {
          cursor: 'default',
        },
        ```

Reply:
Yeah, good call. The logic is a bit gnarly but it works 👍

dbe2c1214c